pages/help/services-resources/index.vue

```diff
@@ -48,29 +36,127 @@ if (!data.value.data && !data.value.helpTopicData) {
   throw createError({ statusCode: 404, message: 'Page not found', fatal: true })
 }
 
+// ELASTIC SEARCH INDEX
+// GETS DATA FROM CRAFT
+// CREATES ES INDEX TO BE SEARCHED
+// CHECK THAT NUXT IS RUNNING ON THE SERVER (process.server)
+console.log('DATA-DATA-DATA-DATA' + data)
+if (
+  data.value.externalResource &&
+  data.value.externalResource.length > 0 &&
+  process.server
+) {
+  for (const externalResource of data.value.externalResource) {
+    await $elasticsearchplugin.index(
+      { ...externalResource, serviceOrResourceType: 'external resource' },
+      externalResource.slug
+    )
+  }
+}
+
 const page = ref(data.value.data)
 const helpTopic = ref(data.value.helpTopicData)
 const summaryData = ref(_get(data.value.data, 'entry', {}))
+
+// ENABLE PREVIEW
 watch(data, (newVal, oldVal) => {
   console.log('In watch preview enabled, newVal, oldVal', newVal, oldVal)
   page.value = _get(newVal.data, 'entry', {})
   helpTopic.value = ref(newVal.helpTopicData)
   summaryData.value = ref(_get(newVal.data, 'entry', {}))
 })
 
+// FUNCTION AND METHODS RELATED TO ES
+// COMPUTED METHODS THAT USE CRAFT DATA
+const hits = ref([])
 const noResultsFound = ref(false)
 const searchFilters = ref([])
-const hits = ref([])
-const searchGenericQuery = ref(
-  {
-    queryText: route.query.q || '',
-    queryFilters:
+
+const searchGenericQuery = ref({
+  queryText: route.query.q || '',
+  queryFilters:
+    (route.query.filters &&
+      JSON.parse(route.query.filters)) ||
+    {},
+})
+
+// THIS WATCHER IS CALLED WHEN THE ROUTER PUSHES UPDATES TO THE QUERY PARAM
+// ie: someone changes the query or starts viewing from a bookmarked page
+watch(() => route?.query, (oldValue, newValue) => {
+  if (oldValue !== newValue) {
+    if (newValue?.q === '') hits.value = []
+    // DO WE NEED THIS LINE BELOW?
+    searchGenericQuery.value.queryText = route.query.q || ''
+    searchES()
+  }
+}, { deep: true, immediate: true })
+
+// ES search function
+async function searchES() {
```

Comment:
replace this function with
async function searchES() {
  if (
    (route.query.q && route.query.q !== '') ||
    (route.query.filters &&
      queryFilterHasValues(
        route.query.filters,
        config.serviceOrResources.filters
      ))
  ) {
    console.log('Search ES HITS query,', route.query.q)
    const queryText = route.query.q || '*'
    const results = await $dataApi.keywordSearchWithFilters(
      queryText,
      config.serviceOrResources.searchFields,
      '(sectionHandle:serviceOrResource OR sectionHandle:workshopSeries OR sectionHandle:helpTopic) OR (sectionHandle:externalResource AND displayEntry:yes)',
      (route.query.filters &&
        JSON.parse(route.query.filters)) ||
      [],
      config.serviceOrResources.sortField,
      config.serviceOrResources.orderBy,
      config.serviceOrResources.resultFields,
      []
    )
    if (results && results.hits && results.hits.total.value > 0) {
      console.log('Search ES HITS,', results.hits.hits)
      hits.value = results.hits.hits
      noResultsFound.value = false
    } else {
      noResultsFound.value = true
      hits.value = []
    }
  } else {
    // console.log('data.value', data.value)
   
    hits.value = []
    noResultsFound.value = false
  }
}

Reply:
we dont need to run graphql queries again in this function like we did in nuxt2, look at news listing index template or collections/access index template for similar implementation , also update jira ticket with updated description.